tutorials/rendering_plugins.md

```diff
@@ -0,0 +1,132 @@
+\page rendering_plugins Rendering plugins
+
+This tutorial will go over how to write Ignition Gazebo plugins that alter the
+3D scene's visual appearance using Ignition Rendering APIs.
+
+This is not to be confused with integrating a new rendering engine. See
+[How to write your own rendering engine plugin](https://ignitionrobotics.org/api/rendering/4.2/renderingplugin.html)
+for that.
+
+This tutorial will go over a couple of example plugins that are located at
+https://github.com/ignitionrobotics/ign-gazebo/tree/main/examples/plugin/rendering_plugins.
+
+## Scenes
+
+During simulation, there are up to two 3D scenes being rendered by
+Ignition Gazebo, one on the server process and one on the client process.
+
+The server-side scene will only be created when using the
+`ignition::gazebo::systems::Sensors` system plugin on the server. This is the
+scene that shows what the sensors see.
+
+The client-side scene will only be created when using the
+`ignition::gazebo::Scene3D` GUI system plugin on the client. This is the
+scene that shows what the user sees.
+
+Ignition Gazebo keeps these scenes in sync by sending periodic state messages
+from the server to the client that contain entity and component data with
+the `ignition::gazebo::systems::SceneBroadcaster` plugin. Any
+changes done to these scenes using Ignition Rendering APIs directly, as
+described in this tutorial, will only affect one of the scenes and will not be
+synchronized.
+
+## Plugin types
+
+Depending on the scene that you want to affect, you'll need to write a
+different plugin.
+
+To interact with the server-side scene, you'll need to write an
+`ignition::gazebo::System`
+See [Create System Plugins](createsystemplugins.html).
+
+To interact with the client-side scene, you'll need to write an
+[ignition::gui::Plugin](https://ignitionrobotics.org/api/gui/4.1/classignition_1_1gui_1_1Plugin.html),
+or a more specialized `ignition::gazebo::GuiSystem`
+if you need to access entities and components.
+See the [GUI system plugin example](https://github.com/ignitionrobotics/ign-gazebo/tree/main/examples/plugin/gui_system_plugin).
+
+## Getting the scene
+
+When writing either plugin type, the `ignition::rendering::Scene` pointer can
+be conveniently found using the rendering engine's singleton. Both example
+plugins use the exact same logic to get the scene:
+
+\snippet examples/plugin/rendering_plugins/RenderingServerPlugin.cc findScene
+
+The function above works for most cases, but you're welcome to customize it
+for your use case.
+
+## Render thread
+
+It's important to keep in mind that all rendering operations should happen in the
+render thread. In order to access that thread from a custom plugin, it's necessary
```

Comment:
Would it be worth mentioning why we have render threads, and/or providing a resource to more information about this in case users want to know more?

Reply:
Added some explanation on b16bb1f, let me know what you think.

Reply:
Looks good, thanks!